applications/tari_app_grpc/proto/types.proto

```diff
@@ -199,6 +199,12 @@ message Signature {
     bytes signature = 2;
 }
 
+// Excess signature identified by hex strings for the public nonce and signature.
+message ExcessSignature {
```

Comment:
Cant you just re-use signature above?
Its the same signature, here you use string, where above it just uses the bytes.

Reply:
So I did try that, and was getting some ugly mismatch where the bytes received by the GRPC server would fail the conversion to the keys

Reply:
That could be where I was using `grpcurl` and probably testing it incorrectly previously - this did make my testing much easier. 

Is this a blocker?

Reply:
There is going to be a str-to-pubkey conversion on the base_node side doing it this way. 
But if the client is not an actual human,like a wallet he is going to use the bytes. 
Thus you will have a pubkey-to-str-to-pubkey conversion. 
To me that sound like we should only do that if required, hence why I would recommend doing the conversion client side before it sends it over, so there is always only one conversion

Reply:
Cool that makes sense - then I'll just need some help figuring out why the keys can't be created from the bytes GRPC is receiving

Reply:
for the provided public nonce `72ae4e7972263c031ce1f1569abcc0e953670bbbefa29c196e4d0afcb46f4819 `, GRPC is receiving this byte array (48 items)

```
[239, 102, 158, 225, 238, 253, 239, 109, 186, 221, 205, 55, 213, 199, 181, 127, 94, 122, 245, 166, 220, 115, 71, 189, 231, 126, 187, 209, 182, 219, 121, 246, 182, 245, 205, 125, 233, 238, 29, 209, 167, 220, 111, 142, 159, 227, 205, 125]
```

where our PublicKey as_bytes function produces this byte array (32 items)

```
[114, 174, 78, 121, 114, 38, 60, 3, 28, 225, 241, 86, 154, 188, 192, 233, 83, 103, 11, 187, 239, 162, 156, 25, 110, 77, 10, 252, 180, 111, 72, 25]
```